src/core/utils/misc.cpp

```diff
@@ -315,30 +317,57 @@ QString Utils::Misc::pythonVersionComplete() {
     return version;
 }
 
-// return best userfriendly storage unit (B, KiB, MiB, GiB, TiB)
+QString Utils::Misc::unitString(Utils::Misc::SizeUnit unit)
+{
+    return QCoreApplication::translate("misc",
+         units[static_cast<int>(unit)].source, units[static_cast<int>(unit)].comment);
+}
+
+// return best userfriendly storage unit (B, KiB, MiB, GiB, TiB, ...)
 // use Binary prefix standards from IEC 60027-2
 // see http://en.wikipedia.org/wiki/Kilobyte
 // value must be given in bytes
 // to send numbers instead of strings with suffixes
-QString Utils::Misc::friendlyUnit(qreal val, bool is_speed)
+bool Utils::Misc::friendlyUnit(qreal rawVal, qreal &val, Utils::Misc::SizeUnit &unit)
```

Comment:
qreal is wrong input type for this function. We expect there some amount of bytes that shouldn't be neither fractional nor negative. So qlonglong will be right here. And return void, of course.